operator/v1/types_metal3provisioning.go

```diff
@@ -0,0 +1,109 @@
+
+	// spec is the specification of the desired behavior of the
+	// Metal3Provisioning.
+	Spec Metal3ProvisioningSpec `json:"spec,omitempty"`
+
+	// status is the most recently observed status of the
+	// Metal3Provisioning.
+	Status Metal3ProvisioningStatus `json:"status,omitempty"`
+}
+
+// ProvisioningDHCP represents just the configuration required to fully
+// identify the way DHCP would be handled during baremetal host bringup.
+//
+// DHCP services could be provided external to the metal3 cluster, in
+// which case, IP address assignment for the baremetal servers should
+// happen via this external DHCP server and not via a DHCP server started
+// within the metal3 cluster.
+// If IP address assignment needs to happen via the DHCP server within the
+// metal3 cluster, then the CR needs to contain the DHCP address range that
+// this internal DHCP server needs to use.
+//
+type ProvisioningDHCP struct {
+	// ManagementState within the OperatorSpec needs to be set to
+	// indicate if the DHCP server is internal or external to the
+	// metal3 cluster. ManagementState set to "Removed" indicates
+	// that the DHCP server is outside the metal3 cluster. And a
+	// value of "Managed" indicates that the DHCP services are
+	// managed within the metal3 cluster.
+	// The other fields of OperatorSpec retain their existing
+	// semantics.
+	OperatorSpec `json:",inline"`
+
+	// If the ManagementState within the OperatorStatus is set to
+	// "Managed", then the DHCPRange represents the range of IP addresses
+	// that the DHCP server running within the metal3 cluster can use
+	// while provisioning baremetal servers. If the value of ManagementState
+	// is set to "Removed", then the value of DHCPRange will be ignored.
+	// If the ManagementState is "Managed" and the value of DHCPRange is
+	// not set, then the DHCP range is taken to be the default range which
+	// goes from .10 to .100 of the ProvisioningNetworkCIDR. This is the only
+	// value in all of the provisioning configuration that can be changed
+	// after the installer has created the CR.
+	DHCPRange string `json:"DHCPRange,omitempty"`
+}
+
+// Metal3ProvisioningSpec is the specification of the desired behavior of the
+// Metal3Provisioning.
+type Metal3ProvisioningSpec struct {
+	// provisioningInterface is the name of the network interface on a Baremetal
+	// server to the provisioning network. It can have values like "eth1" or "ens3".
+	ProvisioningInterface string `json:"provisioningInterface"`
+
+	// provisioningIP is the IP address assigned to the provisioningInterface of
+	// the baremetal server. This IP address should be within the provisioning
+	// subnet, and outside of the DHCP range.
+	ProvisioningIP string `json:"provisioningIP"`
+
+	// provisioningNetworkCIDR is the network on which the baremetal nodes are
+	// provisioned. The provisioningIP and the IPs in the dhcpRange all come from
+	// within this network.
+	ProvisioningNetworkCIDR string `json:"provisioningNetworkCIDR"`
```

Comment:
This isn't required for a first cut.  Let's drop this for now.

Reply:
Sorry what do you mean? This is absolutely critical to have

Reply:
Agreed, we've already pared this down to the absolute minimum in the $many review iterations of https://github.com/openshift/enhancements/pull/119 - at this point we need to implement what was agreed, and IMHO it's not reasonable to start redefining the interface arbitrarily at this point.